tests/xdc/nexus_request_forwarding_test.go

```diff
@@ -382,3 +624,26 @@ func requireExpectedMetricsCaptured(t *testing.T, snap map[string][]*metricstest
 	require.Subset(t, snap["nexus_latency"][0].Tags, map[string]string{"namespace": ns, "method": method, "outcome": expectedOutcome})
 	require.Equal(t, metrics.MetricUnit(metrics.Milliseconds), snap["nexus_latency"][0].Unit)
 }
+
+func (s *NexusRequestForwardingSuite) mustToPayload(v any) *commonpb.Payload {
+	conv := converter.GetDefaultDataConverter()
+	payload, err := conv.ToPayload(v)
+	s.NoError(err)
+	return payload
+}
+
+func (s *NexusRequestForwardingSuite) failoverNamespace(ctx context.Context, ns string) {
+	_, err := s.cluster1.GetFrontendClient().UpdateNamespace(ctx, &workflowservice.UpdateNamespaceRequest{
+		Namespace: ns,
+		ReplicationConfig: &replicationpb.NamespaceReplicationConfig{
+			ActiveClusterName: s.clusterNames[1],
+		},
+	})
+	s.NoError(err)
+
+	// Wait until failover is complete
+	s.Eventually(func() bool {
+		cached, err := s.cluster2.GetHost().GetFrontendNamespaceRegistry().GetNamespace(namespace.Name(ns))
```

Comment:
Nice, we could use this trick in some other tests.

Reply:
Yeah that's why I exposed the frontend namespace registry in another PR. `DescribeNamespace` goest straight to persistence, so it's not useful for waiting until the cache is updated.